source/common/router/scoped_rds.cc

```diff
@@ -267,7 +267,8 @@ void ScopedRdsConfigSubscription::onRdsConfigUpdate(const std::string& scope_nam
   auto new_scoped_route_info = std::make_shared<ScopedRouteInfo>(
       envoy::api::v2::ScopedRouteConfiguration(iter->second->configProto()),
       std::make_shared<ConfigImpl>(rds_subscription.routeConfigUpdate()->routeConfiguration(),
-                                   factory_context_, false));
+                                   factory_context_.getServerFactoryContext(/*is_dynamic=*/false),
```

Comment:
Shouldn't this be dynamic? TBH I'm not a fan of passing this type of param everywhere, but let's discuss in the interface so I can learn more what this is for.